### PR TITLE
feat: caluculate nearest voxel score each point

### PR DESCRIPTION
**Pull request by SaltUhey**
## Description
To visualize matching score each point,  a function is implemented to output a point cloud containing the value of the score of each point.

This score is the highest value of the neighboring voxels of each point.

## Related links
[autoware_universe](https://github.com/autowarefoundation/autoware.universe/pull/8856)

### CODE REVIEW

**Comment by SakodaShintaro** [comment]
@SaltUhey 
The CI check "pre-commit" failed.
Could you apply clang-format to new code?

File: include/multigrid_pclomp/multigrid_ndt_omp.h [hunk 1]
```diff
@@ -554,6 +556,8 @@ class MultiGridNormalDistributionsTransform : public pcl::Registration<PointSour
   boost::optional<Eigen::Matrix4f> regularization_pose_;
   Eigen::Vector3f regularization_pose_translation_;
 
+  pcl::PointCloud<pcl::PointXYZI>::Ptr score_points_{new pcl::PointCloud<pcl::PointXYZI>};
```
Comment (SakodaShintaro) [inline include/multigrid_pclomp/multigrid_ndt_omp.h]: Is this variable necessary?

File: include/pclomp/ndt_struct.hpp [hunk 2]
```diff
@@ -57,6 +57,8 @@
 #include <Eigen/Core>
 #include <Eigen/Geometry>
 
+#include <pcl/common/common.h>
```
Comment (SakodaShintaro) [inline include/pclomp/ndt_struct.hpp]: Is this include necessary?

File: include/multigrid_pclomp/multigrid_ndt_omp.h [hunk 3]
```diff
@@ -262,6 +262,8 @@ class MultiGridNormalDistributionsTransform : public pcl::Registration<PointSour
   // lower is better
   double calculateTransformationProbability(const PointCloudSource & cloud) const;
   double calculateNearestVoxelTransformationLikelihood(const PointCloudSource & cloud) const;
+  pcl::PointCloud<pcl::PointXYZI>::Ptr calculateNearestVoxelScoreEachPoint(
```
Comment (SakodaShintaro) [inline include/multigrid_pclomp/multigrid_ndt_omp.h]: Could you change the return type from `pcl::PointCloud<pcl::PointXYZI>::Ptr` to `pcl::PointCloud<pcl::PointXYZI>`?
Please make the same changes to `multigrid_ndt_omp_impl.hpp`.

**Review by SakodaShintaro — CHANGES_REQUESTED** [review]
Thank you for your contribution!
I left some comments. Please reply to them. :pray:

File: include/multigrid_pclomp/multigrid_ndt_omp.h (hunk 1 above)
Comment (SaltUhey) [reply include/multigrid_pclomp/multigrid_ndt_omp.h]: Thank you, it is not necessary now.

File: include/pclomp/ndt_struct.hpp (hunk 2 above)
Comment (SaltUhey) [reply include/pclomp/ndt_struct.hpp]: Thank you. It was not required.

File: include/multigrid_pclomp/multigrid_ndt_omp.h (hunk 3 above)
Comment (SaltUhey) [reply include/multigrid_pclomp/multigrid_ndt_omp.h]: OK. I fixed it.

**Review by SakodaShintaro — APPROVED** [review]
Thank you for your correction.
I have confirmed that the `logging_simulator` works well and the mean error of `/localization/kinematic_state` is almost the same as before.
Looks Good To Me

**Pull request merged**